File: Frontend-React/src/services/bill.service.ts
import http from '../api/http';

const json = { headers: { 'Content-Type': 'application/json' } };

export const BillService = {
  generateReport: (data: any) => http.post('/bill/generateReport', data, json),
  getPdf: (data: any) => http.post('/bill/getPdf', data, { responseType: 'blob' }),
  getBills: () => http.get('/bill/getBills'),
  delete: (id: any) => http.post('/bill/delete/' + id, json),
  updateOrderStatus: (id: any, status: string) => http.put('/bill/orderStatus/' + id, { status }, json),
  cancelOrder: (id: any) => http.put('/bill/cancel/' + id, {}, json),
};

File: BackEnd/com.inn.cafe/src/main/java/com/inn/cafe/service/BillService.java
package com.inn.cafe.service;

import com.inn.cafe.POJO.Bill;
import com.inn.cafe.dto.OrderStatusUpdateRequest;
import org.springframework.data.domain.Page;
import org.springframework.http.ResponseEntity;

import java.util.List;
import java.util.Map;

public interface BillService {
    ResponseEntity<String> generateReport(Map<String, Object> requestMap);
    ResponseEntity<List<Bill>> getBills();

    ResponseEntity<Page<Bill>> getBillsPaged(int page, int size, String sortBy, String direction);

    ResponseEntity<byte[]> getPdf(Map<String, Object> requestMap);

    ResponseEntity<String> delete(Integer id);

    ResponseEntity<Bill> updateOrderStatus(Integer id, OrderStatusUpdateRequest request);

    ResponseEntity<Bill> cancelOrder(Integer id);
}
